common/statistics/src/gateways.rs

```diff
@@ -0,0 +1,89 @@
+// Copyright 2024 - Nym Technologies SA <contact@nymtech.net>
+// SPDX-License-Identifier: GPL-3.0-only
+
+use nym_credentials_interface::TicketType;
+use nym_sphinx::DestinationAddressBytes;
+use time::OffsetDateTime;
+
+/// Channel for receiving incoming Stats events
+pub type GatewayStatsReceiver = tokio::sync::mpsc::UnboundedReceiver<GatewayStatsEvent>;
+
+/// Channel allowing for generic statistics events to be reported to a stats event aggregator.
+#[derive(Clone)]
+pub struct GatewayStatsReporter {
+    stats_tx: tokio::sync::mpsc::UnboundedSender<GatewayStatsEvent>,
+}
+
+impl GatewayStatsReporter {
+    /// Construct a new gateway statistics event reporter
+    pub fn new(stats_tx: tokio::sync::mpsc::UnboundedSender<GatewayStatsEvent>) -> Self {
+        Self { stats_tx }
+    }
+
+    /// Report a gateway statistivs event using the reporter
+    pub fn report(&self, event: GatewayStatsEvent) {
+        self.stats_tx.send(event).unwrap_or_else(|err| {
+            log::error!("Failed to report gateway stat event : {:?}", err);
```

Comment:
nitpicking: use `Display` impl of the error, i.e. `log::error!("failed to report gateway stat event: {err}");`

Reply:
Fixed in https://github.com/nymtech/nym/pull/5107/commits/0b0f2c77cad0831e84bf88efe3dc94dcf9931faa